### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by Avani8** [lowRISC/opentitan#1194]
1) is there any reference documentation for adding custom IP core?...like aes, hmac examples at sw/device/tests/. if I've my custom component in verilog/system verilog in hw/ip/ and corresponding software implementation at sw/device/lib or at sw/vendor/ (like crypto). 
2) Is there any 32bit linux support available for opentitan?

**Comment by tjaychen** [comment]
Please have a look at our [comportability [](specification](https://docs.opentitan.org/doc/rm/comportability_specification/))

The software story is a bit more in flux.  The general idea is that we want to separate out a baremetal, low level library that other pieces of software can build upon.  I've added a few people who can better answer that question. 

I do not believe anyone has tried porting linux to opentitan thus far.

**Comment by gkelly** [comment]
> Please have a look at our [comportability
> 
> The software story is a bit more in flux. The general idea is that we want to separate out a baremetal, low level library that other pieces of software can build upon. I've added a few people who can better answer that question.
> 
> I do not believe anyone has tried porting linux to opentitan thus far.

OpenTitan doesn't have an MMU, no XIP-capable external memory bus, and a very small amount of RAM.  I think it _might_ be possible to get it running, but we don't have any plans to do that work at this point.

**Comment by tomoveu** [comment]
@Avani8 would adding a SoftCPU on the same FPGA help?

This way you could have CPU running Linux in parallel of OpenTitan, but then you will also hit the issue of memory isolation.

@gkelly just for the test, it should be possible to add some small footprint SoftCPU and fit it on the Artix-7 FPGA together with the OpenTitan?

**Comment by Avani8** [comment]
@tomoveu .sure, adding SoftCPU on the same FPGA helps, but we are planning on ASIC implementation with our custom component and opentitan. Our work does not require full Linux support. But supporting [coreboot](https://github.com/coreboot/coreboot/tree/master/src/arch/riscv)  or [ecos](https://chromium.googlesource.com/chromiumos/platform/ec/+/master/README.md#EC) is what we might try eventually.

@tomoveu I saw your FR regarding PCR like register implementation. We are also interested in that feature. and thanks for explaining things in https://github.com/lowRISC/opentitan/issues/1112(URL)

**Comment by imphil** [comment]
I guess all questions have been answered, closing. Please open a new issue if you have further questions.
